File: proxies/dotnet/Controllers/Spec.cs
using Microsoft.AspNetCore.Mvc;
using Newtonsoft.Json;

namespace dotnet.Controllers;

[ApiController]
[Route("[controller]")]
public class SpecController : ControllerBase
{

    private readonly ILogger<SpecController> _logger;

    public SpecController(ILogger<SpecController> logger)
    {
        _logger = logger;
    }

    [HttpGet]
    public string Get()
    {
        return JsonConvert.SerializeObject(new {
            name = "DotNet",
            version = "",
            capabilities = new List<string>(){"EdgeDB", "LocalBucketing", "CloudBucketing"}
        });
    }
}

File: proxies/java/src/main/java/com/dvc_harness/proxy/models/Spec.java
package com.dvc_harness.proxy.models;

public class Spec {
    private final String name;
    private final String version;
    private final String[] capabilities;

    public Spec() {
        name = "java";
        capabilities = new String[]{"LocalBucketing", "CloudBucketing", "EdgeDB"};
        version = "";
    }

    public String getName() {
        return name;
    }

    public String[] getCapabilities() {
        return capabilities;
    }

    public String getVersion() {
        return version;
    }
}
